### PR TITLE
Convert documentation definition lists to tables

### LINKED ISSUE #14240: Confusing description of options in user guide
Since recently, the user guide uses definition lists to describe new options, more specifically their default and possible values. I feel it confusing.

### Steps to reproduce:
* Go to [Delayed descriptions for characters on cursor movement](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#delayedCharacterDescriptions) paragraph in the User Guide.
* Read and navigate with down arrow or "I" key

### Actual behavior:
* List is reported with 4 item but navigating with I allows to get only two items
* On the contrary to standard lists (`<ul>`), definition list do not present bullets for each item (`<dt>` or maybe `<dl>`); thus it's not easy to understand where the term is and where the description is.
* More generally, the semantic between terms and their definition / description is not present.

### Expected behavior:
The issue with definition list is discussed in #3858. It does not seem that its resolution is a priority and even if it was, the solution proposed in #3858 would not lead to a satisfactory UX when reading the User Guide. Indeed, it is written:
> NVDA should announce "Term" for each dt element.

This would lead to hearing "term option" and "term default" which seems strange. "term" for definition lists, but not for description list.

Since I do not expect #3858 to be resolved soon, I suggest to avoid using definition list in the User Guide. Instead, we may use tables or normal lists.

### NVDA logs, crash dumps and other attachments:
N/A
#### NVDA installed/portable/running from source:
N/A
#### NVDA version:
2022.3
#### Windows version:
N/A
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
N/A
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Not looked at.
#### If NVDA add-ons are disabled, is your problem still occurring?
N/A
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Fixes https://github.com/nvaccess/nvda/issues/14240

### Summary of the issue:
Definition lists are not exposed correctly in NVDA.
They are also not supported in markdown natively.
In order to convert text2tags to markdown, these must be converted to tables

### Description of user facing changes
Definition lists are now tables in the user guide, which are exposed better by NVDA.

The user guide standards have been updated to encourage the following format for documenting settings.
The standards have been expanded to further document settings in a well structured manner.
This is to improve readability, creating a simple standard to learn more about a settings documentation.
This also allows for the future of automatically collating tables into a settings reference similar to key commands.

|||
|-|-|
| Options | Default (Enabled), Enabled, Disabled |
| Default | Enabled |
| Toggle command | ``nvda+shift+e`` |

| Option | Behaviour |
|-|-|
| Enabled | behaviour of enabled |
| Disabled | behaviour of enabled |

### Description of development approach

All t2t files had definition lists converted to markdown.
The following regex was used to perform the conversion.

To remove the definition list terminator `:`
- Find: `((^:.+\n  .*\n)+):`
- replace: `$1`

To replace rows:
- Find: `^: (.*)\n  (.*)\n`
- Replace: `|| $1 | $2 |\n`

### Testing strategy:
Checked built output

### Known issues with pull request:
None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

File: projectDocs/dev/userGuideStandards.md [hunk 1]
```diff
@@ -18,13 +18,17 @@ Feature flags, comboboxes and checkboxes should be included using the following
...
-  Default (Enabled), Enabled, Disabled
-:
+```md
+|||
+|-|-|
+| Options | Default (Enabled), Enabled, Disabled |
+| Default | Enabled |
+| Toggle command | `nvda+shift+e` |
+
+| Option | Behaviour |
```
Comment (CyrilleB79) [inline projectDocs/dev/userGuideStandards.md:28]: This PR should be focused on format conversion to something compatible with markDown, i.e. converting the existing definition lists to tables. Other changes that can lead to discussion and opinion have not been fully discussed (there are unanswered comments); thus I'd really encourage to handle them in a separate PR or issue.

Thus, I'd recommend to remove this new table from User Guide standards.

File: projectDocs/dev/userGuideStandards.md [hunk 2]
```diff
@@ -18,13 +18,17 @@ Feature flags, comboboxes and checkboxes should be included using the following
...
-  Enabled
-: Options
-  Default (Enabled), Enabled, Disabled
-:
+```md
+|||
+|-|-|
+| Options | Default (Enabled), Enabled, Disabled |
+| Default | Enabled |
+| Toggle command | `nvda+shift+e` |
```
Comment (CyrilleB79) [inline projectDocs/dev/userGuideStandards.md:26]: This PR should be focused on format conversion to something compatible with markDown, i.e. converting the existing definition lists to tables. Other changes that can lead to discussion and opinion have not been fully discussed (there are unanswered comments); thus I'd really encourage to handle them in a separate PR or issue.

Thus, I'd recommend to have this addition of the key row discussed in a separate PR or issue. If there is a reason why this should be handled in this same PR, could you please explain in the initial description why?
What happends with toggle keys in the paragraphs describing options where there was no definition list?

File: projectDocs/dev/userGuideStandards.md [hunk 3]
```diff
@@ -18,13 +18,17 @@ Feature flags, comboboxes and checkboxes should be included using the following
...
 Once the anchor is set it cannot be updated, while settings may move categories.
 
-```text2tags
-==== The name of a feature ====[FeatureDescriptionAnchor]
-: Default
-  Enabled
-: Options
-  Default (Enabled), Enabled, Disabled
-:
+```md
```
Comment (seanbudd) [inline projectDocs/dev/userGuideStandards.md:21]:
```suggestion
```md
==== The name of a feature ====[FeatureDescriptionAnchor]

```

File: projectDocs/dev/userGuideStandards.md [hunk 4]
```diff
@@ -18,13 +18,14 @@ Feature flags, comboboxes and checkboxes should be included using the following
...
-  Default (Enabled), Enabled, Disabled
-:
+```t2t
+|| Options | Default (Enabled), Enabled, Disabled |
+|| Default | Enabled |
+|| Toggle command | ``nvda+shift+e`` |
+
+|| Option || Behaviour ||
+| Enabled | behaviour of enabled |
+| Disabled | behaviour of enabled |
```
Comment (michaelDCurran) [inline projectDocs/dev/userGuideStandards.md:28]:
```suggestion
| Disabled | behaviour of disabled |
```

**Review by michaelDCurran — APPROVED** [review]

File: projectDocs/dev/userGuideStandards.md (hunk 2 above)
Comment (seanbudd) [reply projectDocs/dev/userGuideStandards.md]: I've expanded the PR description to document the changes to the standards more.

>What happends with toggle keys in the paragraphs describing options where there was no definition list

When we change standards to the documentation we rarely update all the docs at once. Changes to the standards are just for future work.

**Pull request merged**